tensorrt_llm/_torch/auto_deploy/transform/library/build_model.py

```diff
@@ -49,3 +49,43 @@ def _apply(
         info = TransformInfo(skipped=False, num_matches=1, is_clean=True, has_valid_shapes=True)
 
         return gm, info
+
+
+@TransformRegistry.register("build_and_load_factory_model")
+class BuildAndLoadFactoryModel(BaseTransform):
+    """A simple wrapper transform to build a model via the model factory."""
+
+    def _apply(
+        self,
+        gm: GraphModule,
+        cm: CachedSequenceInterface,
+        factory: ModelFactory,
+        shared_config: SharedConfig,
+    ) -> Tuple[GraphModule, TransformInfo]:
+        # load model with auto sharding
+        assert isinstance(factory, hf.AutoModelFactory)
+        # TODO: figure out correct device map, tp_plan argument here???
+        model = factory.automodel_cls.from_pretrained(
+            factory.model,
+            trust_remote_code=True,
+            device_map="cuda",
+            # tp_plan="auto",
```

Comment:
TODO: not clear to me what is correct here. For example Qwen 2.5 VL with `tp_plan` argument failed but then `device_map="auto"` ignored our world_size...